### PR TITLE
Fix AttributeError on accessing package author (#370)

### PR DESCRIPTION
**Pull request by kam1sh**
# Pull Request Check List

- [x] Added **tests** for changed code.
- [x] Updated **documentation** for changed code.

There is fix for #370, when `poetry build` crashed on accessing incorrect author fields.

Also, I could change `AUTHOR_REGEX` to support for email-only authors, but I don't think there is need to handle those situations. At least it's not crashing right now. 

Or maybe better throw warning on authors like that?

### CODE REVIEW

File: tests/packages/test_package.py [hunk 1]
```diff
@@ -14,3 +14,13 @@ def test_package_authors():
...
+
+
+def test_author_email_only():
+    """
+    Checks Package for correct handling of unparsable authors
+    (email only for example)
+    """
+    package = Package("foo", "0.1.0")
+    package.authors.append("<support@example.com>")
+    assert not package.author_name
```
Comment (drunkwcodes) [inline tests/packages/test_package.py]: May you add `assert package.author_email` accordingly?

Comment (kam1sh) [reply tests/packages/test_package.py]: Thanks for fast reviewing. I added `assert not package.author_email`, because parsing only email would need changing `AUTHOR_REGEX`. 
Of course, I could add something like `author, email = match.author, match.email if match else None, raw_author`, but then there could be situations when someone specified only name instead of email, so idk =/

Comment (drunkwcodes) [reply tests/packages/test_package.py]: Well... it worth changing the regex for this fix.

Comment (kam1sh) [reply tests/packages/test_package.py]: Done! I had do add additional regex pattern, because it's hard to parse `support@example.com` and `<support@example.com>` in one. But that new pattern is easy to understand.

**Comment by kam1sh** [comment]
Sure, tomorrow I add that in tests and try to throw away `re`s, but idk about seventh. I think if there is comma in author, then it's user fault.
Didn't knew about mail.utils, thank you!

**Comment by drunkwcodes** [comment]
It's getting worse.

**"Throwing a custom exception"** makes me feel pleasant comparing to this buggy commit. The second half of the function messed up with the wrong cases to handle.

Not to mention the coding style.

**Comment by kam1sh** [comment]
Not arguing that. So I have to bring back old regex and throw `ValueError` (or something) on pattern mismatch?

I'm striving ¯\_(ツ)_/¯

**Comment by drunkwcodes** [comment]
@kam1sh If I were you, I would make more discussion on the #370.

**Comment by kam1sh** [comment]
After short talk I decided to keep old regex and throw exception with text `{} is not a valid author's name` when it could not be parsed. Also added test for that. 

What's next?

**Comment by drunkwcodes** [comment]
@kam1sh This thread would be a real thing if and only if you fix the regex to let the special characters match as I mentioned at the first review.

**Comment by drunkwcodes** [comment]
@kam1sh This is far from elegant. You should stop now or after next failure.

The regex of author name should keep being a whitelist.
The usage of `email` lib is unnecessary.

And the tests are worse thant not existed.

The PR can't be accepted at all and you kept striving to prove it.

**Comment by kam1sh** [comment]
I don't know what to say.

Fine, I'm closing the PR. Sorry for wasting your time.

**Comment by uSpike** [comment]
> @kam1sh This is far from elegant. You should stop now or after next failure.
> And the tests are worse than not existing.
> The PR can't be accepted at all and you kept striving to prove it.

@drunkwcodes I don't think this is a positive way to interact with someone who is willing to put in the effort of contributing.  I can't speak to the technical aspect of this thread, but I t hink this kind of messaging will turn people off to contributing.  That said, I appreciate that you're taking the time to do code reviews.

**Comment by drunkwcodes** [comment]
@uSpike But it saves time. For subscribers. Please. Fight me.

**Comment by uSpike** [comment]
@drunkwcodes not saying anything is better than 

@kam1sh just FYI you don't have to close this.
> @uSpike Fight me.

No.

@kam1sh feel free to reopen this PR.  The only opinion that matters for this project is @sdispater

**Comment by sdispater** [comment]
@drunkwcodes Please refrain from commenting or reviewing the work of others if you can't be nice or helpful while doing so. This is not the first time I've seen you being rude here so I will only ask this once: please be considerate in your comments or do not bother writing them at all.

@kam1sh You can reopen your PR. I will review it when I have time. I know I am not the most reactive at the moment but I will eventually do it. I have the final say when it comes to accepting or rejecting PRs, so as long as you haven't heard from me you can leave it open.
